mmv1/third_party/terraform/services/tpuv2/resource_tpu_v2_vm_test.go.erb

```diff
@@ -35,63 +30,139 @@ func TestAccTpuV2Vm_update(t *testing.T) {
 				ResourceName:            "google_tpu_v2_vm.tpu",
 				ImportState:             true,
 				ImportStateVerify:       true,
-				ImportStateVerifyIgnore: []string{"zone"},
+				ImportStateVerifyIgnore: []string{"labels", "terraform_labels", "zone"},
+			},
+			{
+				Config: testAccTpuV2Vm_update(context, true),
+			},
+			{
+				ResourceName:            "google_tpu_v2_vm.tpu",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"labels", "terraform_labels", "zone"},
 			},
 			{
-				Config: testAccTpuV2Vm_update(updatedContext),
+				Config: testAccTpuV2Vm_update(context, false),
 			},
 			{
 				ResourceName:            "google_tpu_v2_vm.tpu",
 				ImportState:             true,
 				ImportStateVerify:       true,
-				ImportStateVerifyIgnore: []string{"zone"},
+				ImportStateVerifyIgnore: []string{"labels", "terraform_labels", "zone"},
 			},
 		},
 	})
 }
 
 func testAccTpuV2Vm_full(context map[string]interface{}) string {
 	return acctest.Nprintf(`
-data "google_tpu_v2_runtime_versions" "available" {
+resource "google_tpu_v2_vm" "tpu" {
   provider = google-beta
+
+  name        = "tf-test-test-tpu-%{random_suffix}"
+  zone        = "us-central1-c"
+  description = "Text description of the TPU."
+
+  runtime_version  = "tpu-vm-tf-2.13.0"
+  accelerator_type = "v2-8"
+
+  scheduling_config {
+    preemptible = true
+  }
+
+  data_disks {
+    source_disk = google_compute_disk.disk.id
+    mode        = "READ_ONLY"
+  }
+
+  labels = {
+    foo = "bar"
+  }
+
+  metadata = {
+    foo = "bar"
+  }
+
+  tags = ["foo"]
+
+  lifecycle {
+    prevent_destroy = true
```

Comment:
Why do we want prevent_destroy to be true?

Reply:
This is sort of a trick I've seen used a few times that might be adopted more. The point of this type of test is to create the resource, update it in-place, and then destroy it. However, by default the "update in-place" requirement is not enforced, so changing an immutable field will still pass, and since that will recreate the resource, you can no longer tell if _any_ of the fields are appropriately configured in terms of mutability.

So the workaround you see here ensures that Terraform will error if the resource needs to be recreated, so you know that only in-place updates are taking place if the test passes. Then, in order to clean up after the test is done, we need to set `prevent_destroy = false`, and then finally delete the resource.

Reply:
Longer term we'll be able to use https://github.com/hashicorp/terraform-plugin-testing/issues/61 to assert properties of the test plan